lib/galaxy/model/store/__init__.py

```diff
@@ -803,6 +810,28 @@ def __init__(self):
         self.jobs_by_key = {}
         self.requires_hid = []
 
+    def find_hda(self, input_key: ObjectKeyType) -> Optional[model.HistoryDatasetAssociation]:
+        hda = None
+        if input_key in self.hdas_by_key:
+            hda = self.hdas_by_key[input_key]
+        if input_key in self.hda_copied_from_sinks:
+            hda = self.hdas_by_key[self.hda_copied_from_sinks[input_key]]
+        return hda
```

Comment:
```py
if input_key in self.hdas_by_key:
    return self.hdas_by_key[input_key]
if input_key in self.hda_copied_from_sinks:
    return self.hdas_by_key[self.hda_copied_from_sinks[input_key]]
```
That would be even simpler? However, I assume that `input_key` cannot be in both, so it does *not* matter which one we check first. If it *can* be in both, then, to preserve the logic above, we just switch the if-statements - so if it's in both, we return the hda that's in `hda_copied_from_sinks`.

Same applies for the 2 other find* methods.

Reply:
I'm a big fan of minimizing the number of return statements in a method (https://stackoverflow.com/questions/36707/should-a-function-have-only-one-return-statement/733858#733858) - I don't think this simplification increases readability. But my opinion is pretty subjective I'll readily admit.

Reply:
Oh yes, either way is fine. 
(I trust and lean on McConnell in most cases (the source sited in SO). However, I suspect he wrote this part with more verbose languages in mind, especially VB, which results in `If blah ... Exit Sub  End If` for each early return. He actually says that in 2 cases early returns improve readability - a guard clause is one, when you know the answer is another: "not returning immediately means that you have to write more code" (p. 391). The next page contains an example which mirrors ours: `if cond1: return foo, else if cond2: return bar, return default` 
But that's just a fun academic discussion :-) I'm +1 either way.

Reply:
Awesome - thanks for the detailed review and fun discussion!